src/car_backed_blockstore.rs

```diff
@@ -0,0 +1,272 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! It can often be time, memory, or disk prohibitive to read large snapshots into a database like [`ParityDb`](crate::db::parity_db::ParityDb).
+//!
+//! This module provides an implementer of [`Blockstore`] that simply wraps a [CAR file](https://ipld.io/specs/transport/car/carv1).
+//! **Note that all operations on this store are blocking**.
+//!
+//! On creation, [`CarBackedBlockstore`] builds an in-memory index of the [`Cid`]s in the file,
+//! and their offsets into that file.
+//!
+//! When a block is requested [`CarBackedBlockstore`] scrolls to that offset, and reads the block, on-demand.
+//!
+//! Writes are currently cached in-memory.
+//!
+//! Performance is pathological with random access.
```

Comment:
What do you mean by pathological? Let's clarify it with some numbers.

Reply:
Medical background shows! This is also a valid mathematical term, I would, however, clarify it for posterity's sake.